core/src/main/java/com/amplifyframework/util/Error.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.util;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.Objects;
+
+/**
+ * Utility class to provide helpful functions to use on throwable instances.
+ */
+public final class Error {
```

Comment:
I would expect a class named `Error` to be instantiable, and to represent a particular error.  What about calling this something like `ErrorInspector`, or `ThrowableInspector`, or `ErrorCause` (and rename `containsCause` to just `contains`)?   Just a thought.  I think I like `ErrorInspector` the best, though I don't feel too strongly.

Reply:
I too was uncomfortable with the name `Error` but I didn't want to call it `ErrorUtils` because that's just lame haha. I'll go with `ErrorInspector` as you suggested.